lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,447 @@
+
+    class OpSum(Operator):
+        InputA = InputSlot()
+        InputB = InputSlot()
+
+        Output = OutputSlot()
+
+        def setupOutputs(self):
+            assert self.InputA.meta.getAxisKeys() == list(inner_order)
+            assert self.InputA.meta.shape == self.InputB.meta.shape, "Can't add images of different shapes!"
+            self.Output.meta.assignFrom(self.InputA.meta)
+
+        def execute(self, slot, subindex, roi, result):
+            a = self.InputA.get(roi).wait()
+            b = self.InputB.get(roi).wait()
+            result[...] = a + b
+            return result
+
+        def propagateDirty(self, slot, subindex, roi):
+            self.Output.setDirty(roi)
+
+    @reorder
+    @reorder_options(inner_order, [], outer_order)
+    class LazyOp(Operator):
+        """ test op """
+        name = 'LazyOp'
+
+        in_a = InputSlot(stype=ArrayLike)
+        in_b = InputSlot(stype=ArrayLike)
+
+        out_a = OutputSlot(stype=ArrayLike)
+        out_b = OutputSlot(stype=ArrayLike)
+
+        def __init__(self, inner_order, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.inner_order = inner_order
+            self.opSum = OpSum(parent=self)
+
+        def meth(self, *args):
+            assert isinstance(self, LazyOp)
+            assert len(args) == len(test_args)
+            for test, got in zip(test_args, args):
+                assert test == got
+
+        @classmethod
+        def class_meth(cls, *args):
+            assert isinstance(cls, type)
+            assert len(args) == len(test_args), (args, test_args)
+            for test, got in zip(test_args, args):
+                assert test == got
+
+        @staticmethod
+        def static_meth(*args):
+            assert len(args) == len(test_args)
+            for test, got in zip(test_args, args):
+                assert test == got
+
+        @property
+        def property_meth(self):
+            assert isinstance(self, LazyOp)
+
+        def except_meth(self, *args):
+            assert isinstance(self, LazyOp)
+            assert len(args) == len(test_args)
+            for test, got in zip(test_args, args):
+                assert test == got
+
+        def setupOutputs(self):
+            # out_a pass-through of in_a
+            self.out_a.connect(self.in_a)
+
+            # out_b, sum of in_a and in_b
+            self.opSum.InputA.connect(self.in_a)
+            self.opSum.InputB.connect(self.in_b)
+            self.out_b.connect(self.opSum.Output)
+
+        def propagateDirty(self, slot, subindex, roi):
+            self.out_b.setDirty(roi)
+            if slot == self.in_a:
+                self.out_a.setDirty(roi)
+
+    # single example operator
+    lazyOp = LazyOp(graph=Graph(), inner_order=inner_order)
+    lazyOp.setupOutputs()
+    svg_path = os.path.join(svg_dir, 'lazyOp.svg')
+    generateSvgFileForOperator(svg_path, lazyOp, 5)
+
+    # embed tests in trivial parent operator
```

Comment:
Cool, thx for the more real-worldish example, couldn't we just move this to tests?

Reply:
oh, ok.. you two convinced me!